std/algorithm/sorting.d

```diff
@@ -2994,6 +2994,177 @@ unittest
     }
 }
 
+// medianOf
```

Comment:
Nitpick: I really like this handwritten function, but I had to read the description twice as the name is a bit misleading. For me the median is a _computed_ value or as [Wikipedia](https://en.wikipedia.org/wiki/Median) defines it:

> The median is the value separating the higher half of a data sample, a population, or a probability distribution, from the lower half. In simple terms, it may be thought of as the "middle" value of a data set.

That being said I don't really know a better name and I don't want to spark a huge debate, but I think `swap` in the name could make it more expressive, e.g. `medianSwap`,
`sortedIndexSwap`, `multiOrderedSwap` or `multiIndexSwap`.

FWIW in [this thread](http://forum.dlang.org/thread/n8u980$1gkr$1@digitalmars.com?page=1) it was called `partition5` ;-)